relay_messages.go

```diff
@@ -96,6 +97,7 @@ type lazyCallReq struct {
 	caller, method, delegate, key, as []byte
 
 	arg2StartOffset, arg2EndOffset int
+	arg2                           []byte
```

Comment:
instead of a field, can this be a method that uses the `StartOffset` and `EndOffset` to slice into the payload? Likely not a big performance impact, but want to try and keep the fields here minimal ideally (and right now having the start/end offset as well as the byte buf seems a little redundant)

Reply:
actually i would've preferred to deprecate the offsets completely since it really isn't needed, and we will be adding slices for arg1 and arg2 later for the fragmented relay change. the problem is that the offsets are already exposed through the `CallFrame` interface, and removing them would be a breaking change :( would such breakage be acceptable in this case?

Reply:
Ah, the `CallFrame` interface (and all relay-specific interfaces) are considered internal, and not part of any API contracts. Since the only consumer is Muttley, we can make changes here (we'll just need to update the Muttley code at the same time as the dependency upgrade).

Just check how we're using it in Muttley to ensure we don't need the offset values directly (I think we just need the length which is also fine to get from the `[]byte`).

A micro-optimization note: a slice is 3 values: a data pointer, length, and a capacity. This often means that 2 ints will be more efficient than a slice or `[]byte` in this case. It's likely not going to cause any noticeable performance impact given how many `[]byte` we already store in this case.

Reply:
yea if performance isn't impacted materially, i think going with the more readable/less complex implementation would be more beneficial overall.

Reply:
turns out it's super involved to deprecate the offsets, even just within tchannel-go. will drop the arg2 slice and leave a TODO for later.